check_for_update.go

```diff
@@ -0,0 +1,114 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/coreos/go-semver/semver"
+)
+
+var (
+	downloadPageURL = "https://phraseapp.com/en/cli"
+	releasesURL     = "https://github.com/phrase/phraseapp-client/releases/latest"
+
+	versionCacheFilename = filepath.Join(os.TempDir(), ".phraseapp.version")
+)
+
+func CheckForUpdate() {
+	latestVersion, err := getLatestVersion()
+	if err != nil {
+		printError(err)
+		return
+	}
+
+	if containsAnySub(strings.ToLower(PHRASEAPP_CLIENT_VERSION), []string{"dev", "test"}) {
+		fmt.Printf("You're running a development version (%s) of the PhraseApp client! Latest version is %s.\n", PHRASEAPP_CLIENT_VERSION, latestVersion)
+		return
+	}
+
+	version, err := semver.NewVersion(PHRASEAPP_CLIENT_VERSION)
+	if err != nil {
+		printError(err)
+		return
+	}
+
+	if version.LessThan(*latestVersion) {
+		fmt.Printf("Please consider updating the PhraseApp CLI client (%s < %s)\nYou can get the latest version from %s.\n", version, latestVersion, downloadPageURL)
+	}
+}
+
+func getLatestVersion() (*semver.Version, error) {
+	version, modified, ok := getLatestVersionFromCache()
+
+	if !ok || time.Since(modified) > 24*time.Hour {
+		versionOnline, err := getLatestVersionFromURL()
+		if err == nil {
+			version = versionOnline
+			ioutil.WriteFile(versionCacheFilename, []byte(version.String()), 0600)
+		}
+	}
+
+	return version, nil
+}
+
+func getLatestVersionFromCache() (version *semver.Version, modified time.Time, ok bool) {
```

Comment:
why not just use an `error` instead of `ok bool`? We can always choose to ignore errors when using that function but would not loose the actual error information.